modules/tinymce/src/core/main/ts/caret/CaretContainerRemove.ts

```diff
@@ -55,7 +46,7 @@ const removeTextAndReposition = (caretContainer: Text, pos: CaretPosition): Care
   const text = before.text + after.text;
 
   if (text.length > 0) {
-    setNodeValue(caretContainer, text);
+    deleteZwspChars(caretContainer);
```

Comment:
It just iteratively deletes the ZWSP character within a text node, instead of overwriting the whole text content. This means that if the selection is currently within the text node then it won't lose it's placement, as overwriting the whole content will cause the offset to reset to 0.

As for the node cleanup, that logic would have never run since we know the text length after cleaning up will never be 0.